### DISCUSSION TITLE
[RFC] Example Apps

### DISCUSSION THREAD

**Discussion by merelinguist** [blitz-js/blitz#947] [RFCs] [answered]
## Idea

Now that @aem’s work on recipes/installers has progressed to a relatively stable point, I think we should go all in on adding as many useful recipes as possible. I also think, however, that we should reconsider the value of the `examples` directory. The advantage of recipes over examples is that they eliminate the need for users to manually configure anything complicated in `_document`, `_app`, or pages.

For this reason I think that the examples directory, which is currently a mix of integration and more basic examples, should focus solely on examples of *feature* development with Blitz, rather than, for example, how to set up Tailwind, which I believe is made redundant by recipes.

The store example is a good starting point, but there could be more simple examples, such as

- Blog
- Forum (Reddit-style clone)
- Todo app

These would not only serve as a reference point for the “Blitz-way” to implement certain features, such as comments, or live updates, but would also be a useful starting point for beginners who wanted an even more guided route than generators.

## Implementation

The examples should be built with generators as much as possible, so that the code style and architecture is relatively standardised. Most importantly, they should be well-documented with comments and READMEs, to maximise their educational value. I’d be happy to start building some of these out if the suggestion is approved.

Let’s hear some thoughts!

**Comment by cherta** [comment]
I'm new to Blitz and I'll definitely benefit from looking at well documented sample projects. I'm not new to web development, so I would seek inspiration and get a sense of the Blitz way there than anywhere else.

**Comment by flybayer** [comment, marked as answer]
@merelinguist yeah that sounds perfect to me!!

**Comment by nitaking** [comment]
Indeed, I remember looking for the example directory in this repository.
And I don't think I'm the only one.

I agree, I'd love to help if I could!

**Comment by MrLeebo** [comment]
Will it potentially create churn as the generators/recipes change if the examples will then have to be regenerated to show the latest code? Seems like a "How to create a Reddit clone with Blitz.js" wiki article with instructions could convey the same teaching value without becoming as stale as the recipes evolve.

**Reply by flybayer** [reply]
yeah good point
